packages/components/src/experience-control/index.tsx

```diff
@@ -1,97 +1,126 @@
+	<label
+		className={ clsx( 'experience-control__option', className, {
+			'is-selected': checked,
 		} ) }
-		onClick={ onClick }
 	>
-		<div className="experience-control__button-content">{ children }</div>
-	</Button>
+		<input
+			type="radio"
+			className="experience-control__radio"
+			checked={ checked }
+			onChange={ onClick }
+			value={ value }
+			name={ name }
+			aria-label={ ariaLabel }
+		/>
+		<div className="experience-control__option-content">{ children }</div>
+	</label>
 );
 
-const ExperienceControlBase = ( {
-	label,
-	children,
-	helpText,
-}: {
-	label: string;
+interface ExperienceControlBaseProps {
 	children: React.ReactNode;
-	helpText?: string;
-} ) => (
-	<div className="experience-control">
-		<div className="experience-control__label">{ label }</div>
-		<div className="experience-control__buttons">{ children }</div>
-		{ helpText && <div className="experience-control__help-text">{ helpText }</div> }
-	</div>
-);
+	label?: string;
+	help?: string;
+}
+
+const ExperienceControlBase = ( { children, ...props }: ExperienceControlBaseProps ) => {
+	const { baseControlProps } = useBaseControlProps( props );
+
+	return (
+		<BaseControl className="experience-control" { ...baseControlProps }>
+			<div className="experience-control__options" role="radiogroup">
+				{ children }
+			</div>
+		</BaseControl>
+	);
+};
 
-export function ExperienceControl( {
-	label,
-	onChange,
-	selectedExperience,
-	helpText,
-}: {
+interface ExperienceControlProps {
 	label: string;
-	onChange: ( experience: string ) => void;
-	selectedExperience: string;
-	helpText?: string;
-} ) {
-	const handleChange = ( experience: string ) => {
-		onChange( experience );
-	};
+	onChange: ( experience: ExperienceType ) => void;
+	value: ExperienceType;
+	help?: string;
+}
+
+export function ExperienceControl( { label, onChange, value, help }: ExperienceControlProps ) {
+	const translate = useTranslate();
 
-	const options = [
+	const options: ExperienceOption[] = [
 		{
-			value: Experience.GOOD,
+			value: ExperienceType.GOOD,
 			icon: <IconGood />,
+			ariaLabel: translate( 'Rate as good experience' ),
 		},
 		{
-			value: Experience.NEUTRAL,
+			value: ExperienceType.NEUTRAL,
 			icon: <IconNeutral />,
+			ariaLabel: translate( 'Rate as neutral experience' ),
 		},
 		{
-			value: Experience.BAD,
+			value: ExperienceType.BAD,
 			icon: <IconBad />,
+			ariaLabel: translate( 'Rate as bad experience' ),
 		},
 	];
 
+	const radioGroupName = `experience-control-${ label.toLowerCase().replace( /\s+/g, '-' ) }`;
+
 	return (
-		<ExperienceControlBase label={ label } helpText={ helpText }>
+		<ExperienceControlBase label={ label } help={ help }>
 			{ options.map( ( option ) => (
 				<ExperienceControlOption
 					key={ option.value }
 					className={ `is-${ option.value }` }
-					isSelected={ selectedExperience === option.value }
-					onClick={ () => handleChange( option.value ) }
+					checked={ value === option.value }
```

Comment:
Currently this code is written like it will only be used in controlled mode, but the stories are in uncontrolled mode so it doesn't match. Should we keep it in controlled mode only for now, and change the stories?